File: python/benchmark.py
import sys
import re
import json
from timeit import default_timer as timer

if len(sys.argv) < 3:
    print('Usage: python benchmark.py <input_filename> regex1 regex2 ..')
    sys.exit(1)

def measure(data, pattern):
    start_time = timer()

    regex = re.compile(pattern)
    matches = re.findall(regex, data)

    elapsed_time = timer() - start_time

    print(str(elapsed_time * 1e3) + ' - ' + str(len(matches)))

with open(sys.argv[1]) as file:
    data = file.read()
    # test_regexes = json.loads(sys.argv[2])

    for regex in sys.argv[2:]:
        measure(data, regex)

File: nim/benchmark.nim
import times
import os
import re
import strformat

if paramCount() <= 0:
  echo "Usage: ./benchmark <filename> regex1 regex2 regex3..."
  quit(QuitFailure)

proc measure(data:string, pattern:string) =
  let time = cpuTime()
  let r_pattern = re(pattern)
  let matches: seq[string] = data.findAll(r_pattern)
  let count = len(matches)
  let elapsed_time = cpuTime() - time 
  echo &"{elapsed_time * 1e3} - {count}"

let data = readFile(paramStr(1))


for i in 2..paramCount():
  let pattern = paramStr(i)
  measure(data, pattern)
